manager/repository/persistent/persistent.go

```diff
@@ -0,0 +1,488 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package persistent implements a persistent deployment repository.
+//
+// This package is currently implemented using MondoDB, but there is no
+// guarantee that it will continue to be implemented using MondoDB in the
+// future.
+package persistent
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/kubernetes/deployment-manager/common"
+	"github.com/kubernetes/deployment-manager/manager/repository"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type pDeployment struct {
```

Comment:
This is a persistent deployment. There's a transient deployment named tDeployment in transient.go. Yes, they're in different packages, so they could be the same, but given that we're defining types that differ significantly, I thought it better to give them different names.